src/hotspot/share/code/codeBlob.cpp

```diff
@@ -73,57 +73,32 @@ unsigned int CodeBlob::allocation_size(CodeBuffer* cb, int header_size) {
   return size;
 }
 
-#ifdef ASSERT
-void CodeBlob::verify_parameters() {
-  assert(is_aligned(_size,            oopSize), "unaligned size");
-  assert(is_aligned(_header_size,     oopSize), "unaligned size");
-  assert(is_aligned(_relocation_size, oopSize), "unaligned size");
-  assert(_data_offset <= size(), "codeBlob is too small");
-  assert(code_end() == content_end(), "must be the same - see code_end()");
-#ifdef COMPILER1
-  // probably wrong for tiered
-  assert(frame_size() >= -1, "must use frame size or -1 for runtime stubs");
-#endif // COMPILER1
-}
-#endif
-
-CodeBlob::CodeBlob(const char* name, CodeBlobKind kind, int size, int header_size, int relocation_size,
-                   int content_offset, int code_offset, int frame_complete_offset, int data_offset,
-                   int frame_size, ImmutableOopMapSet* oop_maps, bool caller_must_gc_arguments) :
-  _oop_maps(oop_maps),
-  _name(name),
-  _size(size),
-  _header_size(header_size),
-  _relocation_size(relocation_size),
-  _content_offset(content_offset),
-  _code_offset(code_offset),
-  _frame_complete_offset(frame_complete_offset),
-  _data_offset(data_offset),
-  _frame_size(frame_size),
-  S390_ONLY(_ctable_offset(0) COMMA)
-  _kind(kind),
-  _caller_must_gc_arguments(caller_must_gc_arguments)
-{
-  DEBUG_ONLY( verify_parameters(); )
-}
-
 CodeBlob::CodeBlob(const char* name, CodeBlobKind kind, CodeBuffer* cb, int size, int header_size,
                    int frame_complete_offset, int frame_size, OopMapSet* oop_maps, bool caller_must_gc_arguments) :
   _oop_maps(nullptr), // will be set by set_oop_maps() call
   _name(name),
   _size(size),
-  _header_size(header_size),
   _relocation_size(align_up(cb->total_relocation_size(), oopSize)),
-  _content_offset(CodeBlob::align_code_offset(_header_size + _relocation_size)),
+  _content_offset(CodeBlob::align_code_offset(header_size + _relocation_size)),
   _code_offset(_content_offset + cb->total_offset_of(cb->insts())),
-  _frame_complete_offset(frame_complete_offset),
   _data_offset(_content_offset + align_up(cb->total_content_size(), oopSize)),
   _frame_size(frame_size),
   S390_ONLY(_ctable_offset(0) COMMA)
+  _header_size((uint16_t)header_size),
+  _frame_complete_offset((int16_t)frame_complete_offset),
```

Comment:
Rather than a raw cast, it would be better to use checked_cast here, or better yet, change the incoming parameter types to match the field type.  That way, if the caller is passing a constant, the compiler can check it at compile time.

Reply:
Agree and will do. In all case `sizeof(_Class_)` is used for `header_size`.